host-spec/c04-networking.tm

```diff
@@ -33,36 +33,83 @@
+
     <item><hlink|Protocol Buffers|https://developers.google.com/protocol-buffers/docs/reference/proto3-spec>
   </itemize>
 
-  <subsection|Discovery mechanism>
+  <subsection|Node Identities>
+
+  Each Polkadot Host node maintains a ED25519 key pair which is used to
+  identify the node. The public key is shared with the rest of the network.
+  This allows nodes to establish secure communication channels. Nodes are
+  discovered as described in the Discovery Mechanism Section
+  (<reference|sect-discovery-mechanism>), where each node can be identified
+  by their corresponding <verbatim|PeerId> (<reference|defn-peer-id>).
+
+  Each node must have its own unique ED25519 key pair. Using the same pair
+  among two or more nodes is interpreted as bad behavior.
+
+  <\definition>
+    <label|defn-peer-id>The Polkadot node's <verbatim|PeerId>, formally
+    referred to as <math|P<rsub|id>>, is derived from the ED25519 public key
+    and is structured as defined in the libp2p specification
+    (<slink|https://docs.libp2p.io/concepts/peer-id/>).
+  </definition>
+
+  <subsection|Discovery mechanism><label|sect-discovery-mechanism>
 
-  The Polkadot Host uses varies mechanism to find peers within the network,
+  The Polkadot Host uses various mechanisms to find peers within the network,
   to establish and maintain a list of peers and to share that list with other
   peers from the network.
 
-  The Polkadot Host uses various mechanism for peer dicovery.
-
   <\itemize>
     <item>Bootstrap nodes - hard-coded node identities and addresses provided
-    by network configuration itself. Those addresses are selected an updated
-    by the developers of the Polkadot Host. Node addresses should be selected
-    based on a reputation metric, such as reliability and uptime.
+    by the genesis state specification as described in Appendix
+    <reference|sect-genesis-block>.
 
     <item>mDNS - performs a broadcast to the local network. Nodes that might
-    be listing can respond the the broadcast.
-
-    <item>Kademlia requests - Kademlia supports <verbatim|FIND_NODE>
-    requests, where nodes respond with their list of available peers.
+    be listing can respond the the broadcast. The libp2p mDNS specification
+    defines this process in more detail (<slink|><slink|https://github.com/libp2p/specs/blob/master/discovery/mdns.md>).
+
+    <item>Kademlia requests - Kademlia is a distributed hash table for
+    decentralized networks and supports <verbatim|FIND_NODE> requests, where
+    nodes respond with their list of available peers. Kademlia requests must
+    also contain protocol identifiers as described in Section
+    <reference|sect-protocol-identifier>.
   </itemize>
 
+  <subsubsection|Protocol Identifier><label|sect-protocol-identifier>
+
+  Kademlia nodes only communicate with other nodes using the same protocol
+  identifier. The Polkadot network is identified by <verbatim|dot>
+  (respectively <verbatim|ksmcc3> for Kusama).
```

Comment:
This "protocol identifier" is how we derive the name of the protocols enumerated below. I don't think it's worth mentioning it. However you can indeed mention that Kademlia indeed only communicates with nodes that use the same *protocol name* for Kademlia, namely `/dot/kad` or `/ksmcc3/kad`.